src/optimizer/stats/cost.cpp

```diff
@@ -0,0 +1,209 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// cost.cpp
+//
+// Identification: src/optimizer/stats/cost.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "type/value.h"
+#include "optimizer/stats/selectivity.h"
+#include "expression/comparison_expression.h"
+#include "optimizer/stats/cost.h"
+
+#include <cmath>
+
+namespace peloton {
+namespace optimizer {
+
+//===----------------------------------------------------------------------===//
+// Scan
+//===----------------------------------------------------------------------===//
+double Cost::SingleConditionSeqScanCost(
+  const std::shared_ptr<TableStats>& input_stats,
+  const ValueCondition& condition,
+  std::shared_ptr<TableStats>& output_stats) {
+
+  PL_ASSERT(input_stats != nullptr);
+  PL_ASSERT(condition != nullptr);
+
+  UpdateConditionStats(input_stats, condition, output_stats);
+
+  return input_stats->num_rows * DEFAULT_TUPLE_COST;
+}
+
+double Cost::SingleConditionIndexScanCost(
+  const std::shared_ptr<TableStats>& input_stats,
+  const ValueCondition& condition,
+  std::shared_ptr<TableStats>& output_stats) {
+
+  double index_height = default_index_height(input_stats->num_rows);
+  double index_cost = index_height * DEFAULT_INDEX_TUPLE_COST;
+
+  double selectivity = Selectivity::ComputeSelectivity(input_stats, condition);
+  double scan_cost = selectivity * DEFAULT_TUPLE_COST;
+
+  UpdateConditionStats(input_stats, condition, output_stats);
+
+  return index_cost + scan_cost;
+}
+
+void Cost::CombineConjunctionStats(
+  const std::shared_ptr<TableStats>& lhs,
+  const std::shared_ptr<TableStats>& rhs,
+  const size_t num_rows, const ExpressionType type,
+  std::shared_ptr<TableStats>& output_stats) {
+
+  PL_ASSERT(lhs != nullptr);
+  PL_ASSERT(rhs != nullptr);
+  PL_ASSERT(num_rows > 0);
+
+  size_t num_tuples = 1;
+  double sel1 = lhs->num_rows / static_cast<double>(num_rows);
+  double sel2 = rhs->num_rows / static_cast<double>(num_rows);
+  LOG_TRACE("Conjunction sel1[%f] sel2[%f]", sel1, sel2);
+  switch (type) {
+    case ExpressionType::CONJUNCTION_AND:
+      // (sel1 * sel2) * num_rows
```

Comment:
It's a comment for how we calculate `num_tuples` for conjunction expression instead of dead code.